pkg/scheduler/queue/user_queues.go

```diff
@@ -13,6 +16,16 @@ type Limits interface {
 	// MaxOutstandingPerTenant returns the limit to the maximum number
 	// of outstanding requests per tenant per request queue.
 	MaxOutstandingPerTenant(user string) int
+
+	// ReservedHighPriorityQueriers returns the minimum number of queriers dedicated for high priority
+	// queue per tenant. All queriers still handle priority queue first, but this provides extra protection on
+	// high priority queries from slow normal queries exhausting all queriers.
+	// If ReservedHighPriorityQueriers is capped by MaxQueriersPerUser.
```

Comment:
You meant to say `If ReservedHighPriorityQueriers larger than MaxQueriersPerUser, it is ...`?